docs/proposals/kubemark.md

```diff
@@ -0,0 +1,163 @@
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/kubemark.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Kubemark proposal
+
+## Goal of this document
+
+This document describes a design of Kubemark - a system that allows performance testing of a Kubernetes cluster. It describes the assumption, high level design and discusses possible solutions for lower-level problems. It is supposed to be a starting point for more detailed discussion.
+
+## Current state and objective
+
+Currently performance testing happens on ‘live’ clusters of up to 100 Nodes. It takes quite a while to start such cluster or to push updates to all Nodes, and it uses quite a
+lot of resources. At this scale the amount of wasted time and used resources is still acceptable. In the next quarter or two we’re targeting 1000 Node cluster, which will push
+it way beyond ‘acceptable’ level. Additionally we want to enable people without many resources to run scalability tests on bigger clusters
+than they can afford at given time. Having an ability to cheaply run scalability tests will enable us to run some set of them on "normal"
+test clusters, which in turn would mean ability to run them on every PR.
+
+This means that we need a system that will allow for realistic performance testing on (much) smaller number of “real” machines. First assumption we make is that Nodes are
+independent, i.e. number of existing Nodes do not impact a performance of a single Node. This is not entirely true, as number of Nodes can increase latency of various
+components on Master machine, which in turn may increase latency of Node operations, but we’re not interested in measuring this effect here. In turn we want to measure how
+number of Nodes and the load imposed by Node daemons affects the performance of Master components.
+
+## Kubemark architecture overview
+
+The high-level idea behind Kubemark is to write library that allows running artificial "Hollow" Nodes that will be able to simulate a behavior of real Kubelet and KubeProxy in a single, lightweight
+binary. Hollow components will need to correctly respond to Controllers (via API server), and preferably, in the fullness of time, be able to ‘replay’
+previously recorded real traffic (this is out of scope for initial version). In the initial version we only want them to be able to fool Master
+components and put some configurable (in what way TBD) load on them. To teach Hollow components replaying recorded traffic they will need
```

Comment:
It seems that the sentence "To teach Hollow components..." and "Such data can be..." should go right before the sentence "In the initial version..." because they are describing the "fullness of time" version not the initial version.